core/src/wallet/ethereum/ERC20/ERC20LikeAccount.cpp

```diff
@@ -126,6 +190,47 @@ namespace ledger {
             return result;
         }
 
+        std::vector<std::shared_ptr<api::ERC20LikeOperation>>
+        ERC20LikeAccount::getSortedOperationsRange(
+            const std::chrono::system_clock::time_point& endDate
+        ) {
+            auto localAccount = _account.lock();
+            soci::session sql (localAccount->getWallet()->getDatabase()->getPool());
+            soci::rowset<soci::row> rows = (sql.prepare << "SELECT op.uid, op.ethereum_operation_uid, op.account_uid,"
+                    " op.type, op.hash, op.nonce, op.value,"
+                    " op.date, op.sender, op.receiver, op.input_data,"
+                    " op.gas_price, op.gas_limit, op.gas_used, op.status"
+                    " FROM erc20_operations AS op"
+                    " WHERE op.account_uid = :account_uid AND op.date <= ':end_date_uid'"
+                    " ORDER BY op.date ASC", soci::use(_accountUid), soci::use(endDate));
+            std::vector<std::shared_ptr<api::ERC20LikeOperation>> result;
+            for (auto& row : rows) {
```

Comment:
Could you please refacto this code with the one used in `ERC20LikeAccount::getOperations` ?

Reply:
I still think this is a bit a pity to throw away some code that would reduce the network’s bandwidth, the client’s CPU time spending sorting data, etc., solely for refactoring concerns, but seems @teams2ua and you seem to have made your mind about that and that I’m the only one thinking otherwise, I’ll remove that code and switch to `getOperations` then.

Reply:
I fill you pain, but I believe this change will allow us to have even more improved performance in the future, if we will need this. And it gives us less code (=less bugs) right now.